### PR TITLE
[WIP] Auto Mission Assistant

### PR DESCRIPTION
**Pull request by TShapinsky**
capture
# Revised:
capture3

## TODO
- [ ] have pause button hold craft without switching out of auto mode
- [ ] make next, previous, and restart, function after mission is completed

### CODE REVIEW

**Comment by TShapinsky** [comment]
fixes  #754

File: Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java [hunk 1]
```diff
@@ -1,17 +1,278 @@
...
+                break;
+            }
+			case R.id.mc_prev: {
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = mission.getCurrentMissionItem() - 1;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
+                break;
+            }
+			case R.id.wp_speed_plus:
+                changeWaypointSpeedBy(100.0);
```
Comment (squilter) [inline Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java]: have you tried using do_change_speed()?

File: Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java [hunk 2]
```diff
@@ -1,17 +1,278 @@
...
+                drone.pauseAtCurrentLocation();
+                break;
+            }
+			case R.id.mc_restart: {
+                if(mission.getCurrentMissionItem() == mission.getMissionItems().size()){
+                    resetAutoMode();
+                }
+                msg_mission_set_current msg = new msg_mission_set_current();
+                msg.seq = 0;
+                ExperimentalApi.sendMavlinkMessage(drone, new MavlinkMessageWrapper(msg));
```
Comment (squilter) [inline Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java]: This command doesn't work yet.  See diydrones/ardupilot#2584.  It isn't on the list to be fixed with 3.4.  So maybe we should remove this button for now, or find a workaround.

Comment (TShapinsky) [inline Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java]: it works as long as the mission hasn't finished yet, see TODO list

File: Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java (hunk 1 above)
Comment (TShapinsky) [inline Android/src/org/droidplanner/android/fragments/mode/ModeAutoFragment.java]: @squilter I have and it either does not work in auto mode or is not implemented by ArduCopter

**Comment by ne0fhyk** [comment]
@TShapinsky let's get rid of the speed control buttons.

**Comment by ne0fhyk** [comment]
@TShapinsky let's also replace the icons with ones from the material design set: https://www.google.com/design/icons/

For the progress bar, we can use something like [this](https://github.com/daimajia/NumberProgressBar) to indicate the mission completion percentage value alongside the progress bar.

**Comment by ne0fhyk** [comment]
@TShapinsky and let's drop the labels: `Mission progress`, `Mission control`.

**Comment by TShapinsky** [comment]
@ne0fhyk why is it important to show the numerical percentage completeness of the mission? I'm also hesitant to add another custom ui element. I can make a progressbar, if needed, that will be more material compliant than the one you linked.

**Comment by TShapinsky** [comment]
changing branch to merge into

**Pull request closed without merging**